cmd/telemeter-client/main.go

```diff
@@ -301,30 +207,115 @@ func (o *Options) Run() error {
 	transformer.With(metricfamily.TransformerFunc(metricfamily.PackMetrics))
 	transformer.With(metricfamily.TransformerFunc(metricfamily.SortMetrics))
 
-	worker := forwarder.New(*from, toUpload, transformer, o.Rules...)
-	worker.ToClient = metricsclient.New(toClient, o.LimitBytes, o.Interval, "federate_to")
-	worker.FromClient = metricsclient.New(fromClient, o.LimitBytes, o.Interval, "federate_from")
-	worker.Interval = o.Interval
+	cfg := forwarder.Config{
+		From:          from,
+		ToAuthorize:   toAuthorize,
+		ToUpload:      toUpload,
+		FromToken:     o.FromToken,
+		ToToken:       o.ToToken,
+		FromTokenFile: o.FromTokenFile,
+		ToTokenFile:   o.ToTokenFile,
+		FromCAFile:    o.FromCAFile,
+
+		AnonymizeLabels:   o.AnonymizeLabels,
+		AnonymizeSalt:     o.AnonymizeSalt,
+		AnonymizeSaltFile: o.AnonymizeSaltFile,
+		Interval:          o.Interval,
+		LimitBytes:        o.LimitBytes,
+		Rules:             o.Rules,
+		RulesFile:         o.RulesFile,
+		Transformer:       transformer,
+	}
+
+	worker, err := forwarder.New(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to configure Telemeter client: %v", err)
+	}
 
 	log.Printf("Starting telemeter-client reading from %s and sending to %s (listen=%s)", o.From, o.To, o.Listen)
 
-	go worker.Run()
+	var g run.Group
+	{
+		// Execute the worker's `Run` func.
+		ctx, cancel := context.WithCancel(context.Background())
+		g.Add(func() error {
+			worker.Run(ctx)
+			return nil
+		}, func(error) {
+			cancel()
+		})
+	}
+
+	{
+		// Notify and reload on SIGHUP.
+		hup := make(chan os.Signal)
```

Comment:
nit, according to https://godoc.org/os/signal#Notify:
> Package signal will not block sending to c: the caller must ensure that c has sufficient buffer space to keep up with the expected signal rate. For a channel used for notification of just one signal value, a buffer of size 1 is sufficient.

Hence, I suggest to `hup := make(chan os.Signal, 1)`

Reply:
Great suggestion!